samples: nrf9160: cloud_client: Fix issues for release 1.4.0

simensrostad opened the pull request:
This PR:
 - Adds simple cloud connect work item that tries to reconnect upon disconnect from cloud. Configurable re-connection intervals.
 - Adds list of possible options to ```CONFIG_CLOUD_BACKEND``` Kconfig.
 - Adds connection flag that allows/aborts cloud publications depending on if the client is connected or not.
 - List the most basic cloud configurations for the different cloud backends in the ```prj.conf```. This makes it more intuitive what options must be set in order to make the respective cloud backend function with the sample.

NCSDK-6623

rlubos commented on samples/nrf9160/cloud_client/src/main.c:
```diff
@@ -60,12 +78,23 @@ void cloud_event_handler(const struct cloud_backend *const backend,
 	case CLOUD_EVT_DISCONNECTED:
 		LOG_INF("CLOUD_EVT_DISCONNECTED");
+
+		cloud_connected = false;
```
nit: This variable is introduced in the following commit, so it will break bisectability. Would be good to move this line to https://github.com/nrfconnect/sdk-nrf/pull/3194/commits/18f6855318e605a4565eaa011ae2e7b442ebb226

simensrostad replied:
Done, thanks.

ru-fu commented on samples/nrf9160/cloud_client/README.rst:
```diff
@@ -45,7 +48,7 @@ The sample supports the following development kits:
 *****
 
-For configuring the different cloud backends, refer to the documentation on :ref:`lib_nrf_cloud` and :ref:`lib_aws_iot`.
+For configuring the different cloud backends, refer to the documentation on :ref:`lib_nrf_cloud`, :ref:`lib_aws_iot` and :ref:`lib_azure_iot_hub`.
```
```suggestion
For configuring the different cloud backends, refer to the documentation on :ref:`lib_nrf_cloud`, :ref:`lib_aws_iot`, and :ref:`lib_azure_iot_hub`.
```

ru-fu commented on samples/nrf9160/cloud_client/README.rst:
```diff
@@ -18,6 +18,7 @@ The current version of the sample supports the following libraries as cloud back
 
 *  :ref:`lib_nrf_cloud`
 *  :ref:`lib_aws_iot`
+*  :ref:`lib_azure_iot_hub'
```
```suggestion
*  :ref:`lib_azure_iot_hub`
```

ru-fu commented on samples/nrf9160/cloud_client/Kconfig:
```diff
@@ -9,6 +9,8 @@ menu "Cloud Client Sample Settings"
 	string "Selects the cloud backend to be used"
 	default "NRF_CLOUD"
+	help
+	  Possible values are "NRF_CLOUD", "AWS_IOT", "AZURE_IOT_HUB"
```
```suggestion
	  Possible values are "NRF_CLOUD", "AWS_IOT", "AZURE_IOT_HUB".
```

ru-fu commented on samples/nrf9160/cloud_client/Kconfig:
```diff
@@ -18,6 +20,10 @@ config CLOUD_MESSAGE_PUBLICATION_INTERVAL
 	default 10
 
+config CLOUD_CONNECTION_RETRY_TIMEOUT_SECONDS
+	int "Number of seconds between each cloud connection retry"
```
```suggestion
	int "Sets the number of seconds between each cloud connection retry"
```

ru-fu reviewed (commented):
Please see my fixes - line 21 in the readme actually breaks the doc build.

If these issues are fixed, the docs look good.

jtguggedal reviewed (approved):
Looks good to me

simensrostad commented:
> Please see my fixes - line 21 in the readme actually breaks the doc build.
> 
> If these issues are fixed, the docs look good.

I have addressed your issues. Thank you.
